boards/arm64/qemu_cortex_a53: change to 4 cpus

laroche opened the pull request:
It would be nice to increase the number of CPUs for Qemu-Cortex-A53 to test
more of the SMP functionality.

best regards,

Florian La Roche

npitre commented:
Might be worth removing files from tests//kernel/smp/boards/ at the same 
time as they would become redundant.

laroche committed:
boards/arm64/qemu_cortex_a53: change to 4 cpus

Add 2 more cpus to the device tree and set the default to max 4 cpus.
Remove override for smp tests that updated to 4 cpus.

Signed-off-by: Florian La Roche <Florian.LaRoche@gmail.com>

laroche commented:
Good catch. I've updated the patch to also remove tests//kernel/smp/boards.

best regards,

Florian La Roche

TaiJuWu reviewed (approved):
LGTM

stephanosio commented on dts/arm64/qemu/qemu-virt-a53.dtsi:
```diff
@@ -37,6 +37,18 @@
 			compatible = "arm,cortex-a53";
 			reg = <1>;
 		};
+
+		cpu@2 {
+			device_type = "cpu";
+			compatible = "arm,cortex-a53";
+			reg = <2>;
+		};
+
+		cpu@3 {
+			device_type = "cpu";
+			compatible = "arm,cortex-a53";
+			reg = <3>;
+		};
```
`qemu-virt-a53.dtsi` is included by both `qemu_cortex_a53.dts` and `qemu_cortex_a53_smp.dts`. With this, now we have 4 CPU devicetree nodes in `qemu_cortex_a53` that only has one CPU.

cefba05a71af8d2b10c2eace1f082d9e595c59c0 is incorrect and should be reverted -- this only "worked" because it is not currently used to infer anything.

The additional CPU nodes should be added in `qemu_cortex_a53_smp.dts`.
